bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -0,0 +1,139 @@
+ *                --bigtableInstanceId=$INSTANCE \
+ *                --bigtableTableId=$TABLE \
+ *                --hbaseSnapshotSourceDir=gs://$HBASE_EXPORT_ROOT_PATH \
+ *                --snapshotName=$SNAPSHOT_NAME  \
+ *                --restoreDir=gs://$RESTORE_PATH
+ * </pre>
+ *
+ * Note: restoreDir is a GCS path used to save temporal files generated by Snapshot scans. and it
+ * must not be a sub-folder of the hbaseSnapshotSourceDir.
+ */
+@InternalExtensionOnly
+public class ImportJobFromHbaseSnapshot {
+  private static final Log LOG = LogFactory.getLog(ImportJobFromHbaseSnapshot.class);
+
+  public interface ImportOptions extends ImportJob.ImportOptions {
+    @Description("The HBase root dir where HBase snapshot files resides.")
+    String getHbaseSnapshotSourceDir();
+
+    @SuppressWarnings("unused")
+    void setHbaseSnapshotSourceDir(String hbaseSnapshotSourceDir);
+
+    @Description("Temporal location for restoring snapshots")
+    String getRestoreDir();
+
+    @SuppressWarnings("unused")
+    void setRestoreDir(String restoreDir);
+
+    @Description("Snapshot name")
+    String getSnapshotName();
+
+    @SuppressWarnings("unused")
+    void setSnapshotName(String snapshotName);
+  }
+
+  public static void main(String[] args) {
+    PipelineOptionsFactory.register(ImportOptions.class);
+
+    ImportOptions opts =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(ImportOptions.class);
+
+    LOG.info("DEBUG===> Building Pipeline");
+    Pipeline pipeline = buildPipeline(opts);
+
+    LOG.info("DEBUG===> Running Pipeline");
+    PipelineResult result = pipeline.run();
+
+    if (opts.getWait()) {
+      Utils.waitForPipelineToFinish(result);
+    }
+  }
+
+  @VisibleForTesting
+  static Pipeline buildPipeline(ImportOptions opts) {
+
+    Pipeline pipeline = Pipeline.create(Utils.tweakOptions(opts));
+    try {
+      Configuration configuration =
+          new HBaseSnapshotInputConfigBuilder()
+              .setProjectId(opts.getProject())
+              .setHbaseSnapshotSourceDir(opts.getHbaseSnapshotSourceDir())
+              .setSnapshotName(opts.getSnapshotName())
+              .setRestoreDir(opts.getRestoreDir())
+              .build();
+      pipeline
+          .apply(
+              "Read from HBase Snapshot",
+              HadoopFormatIO.<ImmutableBytesWritable, Result>read()
+                  .withConfiguration(configuration))
+          .apply("Create Mutations", ParDo.of(new HBaseResultToMutationFn()))
+          .apply("Write to Bigtable", createSink(opts));
+
+    } catch (Exception e) {
+      LOG.fatal("Failed to create HBaseConfiguration for HadoopFormatIO");
+      LOG.fatal(e);
+      System.exit(-1);
+    }
+    return pipeline;
+  }
+
+  static PTransform<PCollection<Mutation>, PDone> createSink(ImportOptions opts) {
+    CloudBigtableTableConfiguration config = TemplateUtils.BuildImportConfig(opts);
+    return CloudBigtableIO.writeToTable(config);
+  }
```

Comment:
I'm not sure if this helper is necessary...why not just inline it?

Reply:
Done. 

There was an extra configuration we used to workaround the bulk flush issue, now it is not needed.